Update finding and installing plugin docs (#541)
napari/docs: docs/plugins/start_using_plugins/finding_and_installing_plugins.md

Reviewer: This is great. What comes to mind as missing is mention of how to upgrade plugins. I think, by default, a user working through npm has to uninstall and then reinstall the version they want. Is that correct? Is this the recommended way for most users?

Reviewer 2: the new plugin manager should update plugins too.

Reviewer: Automagically? I remember there used to be an 'upgrade' button or something, but on main that doesn't seems to be the case -- just uninstall is available (the plugin here has v0.9.7 available through npm, I intentionally used npm to download v0.9.5):
![image](https://github.com/user-attachments/assets/61267920-12f3-4657-8868-d2e5e7f1a865)

```
napari: 0.5.6.dev10+g224f8e9c
Platform: Windows-11-10.0.22631-SP0
Python: 3.12.8 | packaged by conda-forge | (main, Dec 5 2024, 14:06:27) [MSC v.1942 64 bit (AMD64)]
Qt: 5.15.2
PyQt5: 5.15.11
NumPy: 1.26.4
SciPy: 1.14.1
Dask: 2024.12.1
VisPy: 0.14.3
magicgui: 0.10.0
superqt: 0.7.0
in-n-out: 0.2.1
app-model: 0.3.1
psygnal: 0.11.1
npe2: 0.7.7
pydantic: 2.10.4

OpenGL:
- GL version: 4.6.0 NVIDIA 551.78
- MAX_TEXTURE_SIZE: 32768
- GL_MAX_3D_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1920x1080, scale 2.0
- screen 2: resolution 1440x2560, scale 1.0

Optional:
- numba: 0.60.0
- triangle: 20230923
- napari-plugin-manager: 0.1.3

Settings path:
- C:\Users\timmo\AppData\Local\napari\napari-testing_e9e6cb9e948feb28aa4661b58255680aebaae620\settings.yaml
```

Reviewer 2: Um it should work?
https://napari.org/napari-plugin-manager/index.html#updating-a-plugin
Maybe there's been a regression somewhere or it's another conda vs pip difference?

# Finding and installing plugins

Information about available plugins can be found in the following places:
- **PyPI:** napari plugins are Python packages distributed on the Python Package Index
  (PyPI), and annotated with the
  tag [`Framework :: napari`](https://pypi.org/search/?q=&o=&c=Framework+%3A%3A+napari) 
- **conda-forge:** You can search for napari plugins from
  the conda-forge [package search page](https://conda-forge.org/packages/).

> - **napari hub:** The [napari hub](https://napari-hub.org) uses this data, together with additional

Reviewer: 'This data' is ambiguous. I'm not sure how necessary it is. Maybe just something like' The napari-hub uses plugin-provided metadata to produce a more user-friendly way to find napari plugins.

Reviewer 3: ```suggestion
- **napari hub:** The [napari hub](https://napari-hub.org) uses package data, together with additional
```

>   metadata, to produce a more user-friendly way to find napari plugins.

Reviewer 3: ```suggestion
  metadata, to produce a user-friendly way to find napari plugins.
```

> Similarly, the napari application lists plugins, annotated on PyPI with `Framework :: napari`,
> in the `Plugins > Install/Uninstall Plugins` menu.

Reviewer: ```suggestion

```

Reviewer: Redundant with below section, but especially with `testing_and_publish/deploy.md`

## Installing plugins with napari

All PyPI packages annotated with the `Framework :: napari` tag can be installed

> directly from within napari, using the [napari plugin manager](https://napari.org/napari-plugin-manager/).

Reviewer: Perhaps it should be stated that npm is only installed with `pip install napari[all]`, `conda install napari`, and the bundled app?

Reviewer 2: Oh this is a good point! The plugin manager has been devolved, so it's not in napari/napari anymore.

Reviewer 3: ```suggestion
directly from within napari, when the user has installed the [napari plugin manager](https://napari.org/napari-plugin-manager/).
```

> To install a new plugin, click the "Plugins" menu and select "Install/Uninstall Plugins...".

Reviewer 3: ```suggestion
After the napari plugin manager is available, install a plugin by clicking the "Plugins" menu and selecting "Install/Uninstall Plugins...".
```

Then, in the plugin installer menu that opens, you can either:

![napari viewer's Plugins menu with Install/Uninstall Plugins as the first item.](../../_static/images/plugin-menu.png)

1. Install plugins by searching: you can start typing in the text box at the top of the dialog to filter plugins. From
   the resulting list, you can then install a plugin by clicking the `Install` button in the tile of the plugin you want
   to install. You can also uninstall or update plugins from this dialog in a similar way under the "Installed Plugins"
   section.

2. Install plugins via manual input: at the bottom of the napari plugin manager window, there is a text box that will

>    say either "install with 'pip' by name/url, or drop file...", or "install with 'pip' by name/url, or drop file..."

Reviewer: I assume this is meant to say 'conda' or something for one of them? Right now both say 'pip'. (my npm is currently not connecting in any install, no idea why, so I can't check)

Author: oops yes, wrong copy/paste!

>    say either "install with 'pip' by name/url, or drop file...", or "install with 'pip' by name/url, or drop file..."

Reviewer 3: ```suggestion
   say either "install with 'pip' by name/url, or drop file...", or "install with 'conda' by name/url, or drop file..."
```

    [requirement specifier](https://pip.pypa.io/en/stable/reference/requirement-specifiers/) or
    [VCS scheme](https://pip.pypa.io/en/stable/topics/vcs-support)). Then, click the "Install" button next to the input
    bar. This method allows for more flexibility in the plugins you can install, but it is also a bit more advanced.

   ![napari viewer's Plugin dialog. At the bottom of the dialog, there is a place to install by name, URL, or dropping in a file.](../../_static/images/plugin-install-dialog.png)

   ```{admonition} Example
   To install `napari-svg`, enter `napari-svg` in the text field and press {kbd}`Enter` or click "Install". This is equivalent to running `pip install napari-svg`.
   ```
   ```{admonition} Example
   If you want to install `napari-svg` directly from the development branch on the [github repository](https://github.com/napari/napari-svg), enter `git+https://github.com/napari/napari-svg.git` in the text field.
   ```
   ```{admonition} Example
   If you want to install `napari-svg` from a specific release, enter `napari-svg==0.1.0` in the text field.
   ```

> 3. Advanced installation: after you have searched for the plugin you with to install, clicking on "Installation Info"

Reviewer: ```suggestion
3. Advanced installation: after you have searched for the plugin you wish to install, clicking on "Installation Info"
```
